services/brig/src/Brig/Calling/API.hs

```diff
@@ -124,14 +126,30 @@ newConfig env mSftEnv limit = do
   srvs <- for finalUris $ \uri -> do
     u <- liftIO $ genUsername tTTL prng
     pure $ Public.rtcIceServer (uri :| []) u (computeCred sha secret u)
-  sftSrvEntries <- maybe (pure Nothing) ((fmap discoveryToMaybe) . readIORef . sftServers) mSftEnv
-  -- According to RFC2782, the SRV Entries are supposed to be tried in order of
-  -- priority and weight, but we internally agreed to randomize the list of
-  -- available servers for poor man's "load balancing" purposes.
-  -- FUTUREWORK: be smarter about list orderding depending on how much capacity SFT servers have.
-  randomizedSftEntries <- liftIO $ mapM randomize sftSrvEntries
-  pure $ Public.rtcConfiguration srvs (sftServerFromSrvTarget . srvTarget <$$> randomizedSftEntries) cTTL
+  sftEntries <- case mSftEnv of
+    Nothing -> pure Nothing
+    Just actualSftEnv -> do
+      sftSrvEntries <- fmap discoveryToMaybe . readIORef . sftServers $ actualSftEnv
+      -- According to RFC2782, the SRV Entries are supposed to be tried in order of
+      -- priority and weight, but we internally agreed to randomize the list of
+      -- available servers for poor man's "load balancing" purposes.
+      -- FUTUREWORK: be smarter about list orderding depending on how much capacity SFT servers have.
+      randomizedSftEntries <- liftIO $ mapM randomize sftSrvEntries
+
+      -- Currently (Sept 2020) the client initiating an SFT call will try all
+      -- servers in this list. Limit this list to a smaller subset in case many
+      -- SFT servers are advertised in a given environment.
+      let subsetLength = Calling.sftListLength actualSftEnv
+      return $ subsetSft subsetLength <$> randomizedSftEntries
```

Comment:
Since all of this is _pure-ish_, we can unit test this.

Reply:
Let's maybe discuss in a call how best to test this without using an integration test; as this happens outside the easily-unit-testable discoveryloop logic.